python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -528,100 +562,275 @@ def asset_key_for_input(self, input_name: str) -> AssetKey:
         else:
             return key
 
+    ### Partition methods
+
     @public
-    def asset_partition_key_for_output(self, output_name: str = "result") -> str:
-        """Returns the asset partition key for the given output. Defaults to "result", which is the
-        name of the default output.
+    def partition_key_for_asset(
+        self, asset: CoercibleToAssetKey, is_dependency: bool = False
```

Comment:
the `is_dependency` bool param here is to handle the self-dependent partition case. Basically when you have a self-dependent partition, if you said `context.partition_key_for_asset("my_asset")` it's unclear if you want the partition key of `my_asset` as a dependency, or as is being currently materialized. I'm handling this (for now at least) by allowing users to specify `is_dependency` to indicate they want it loaded as a dependency.

In all other cases, the function should be able to figure out the correct partition key to load given the current asset

I'm not set on this solution, so if you have any concerns or other ideas let me know